src/scripts/main.js

```diff
@@ -1,3 +1,385 @@
+    tile.setXY(this.x, this.y);
+    this.mergedTile = tile;
+  }
+
+  mergeTiles() {
+    this.tile.setValue(this.tile.value + this.mergedTile.value);
+    this.tile.addMergeAnimation();
+
+    this.mergedTile.removeFromDom();
+    this.mergedTile = null;
+  }
+
+  unlinkTile() {
+    this.tile = null;
+  }
+}
+
+class Tile {
+  constructor(gridElement) {
+    this.tileElement = document.createElement('div');
+    this.value = 0;
+    this.setValue(Math.random() > 0.8 ? 4 : 2);
+    gridElement.append(this.tileElement);
+  }
+
+  setXY(x, y) {
+    this.x = x;
+    this.y = y;
+
+    this.tileElement.style.setProperty('--x', `${this.x}`);
+    this.tileElement.style.setProperty('--y', `${this.y}`);
+  }
+
+  setValue(value) {
+    this.value = value;
+    this.tileElement.classList = [];
+    this.tileElement.classList.add('tile');
+    this.tileElement.classList.add(`tile--${this.value}`);
+    this.tileElement.textContent = `${this.value}`;
+  }
+
+  removeFromDom() {
+    this.tileElement.remove();
+  }
+
+  addMergeAnimation() {
+    this.tileElement.classList.add(`tile--merge`);
+  }
+}
+
+const gameBoard = document.getElementById('game-board');
+const startButton = document.querySelector('.button.start');
+const scoreElement = document.querySelector('.game-score');
+const winMessage = document.querySelector('.message-win');
+const loseMessage = document.querySelector('.message-lose');
+
+startButton.addEventListener('click', startGame);
+
+const grid = new Grid(gameBoard);
+
+let score = 0;
+
+function startGame() {
+  resetGame();
+}
+
+function resetGame() {
+  const allTiles = gameBoard.querySelectorAll('.tile');
+
+  grid.cells.forEach((cell) => (cell.tile = null));
+  allTiles.forEach((tile) => tile.remove());
+
+  grid.getRandomEmptyCell().linkTile(new Tile(gameBoard));
+  grid.getRandomEmptyCell().linkTile(new Tile(gameBoard));
+
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+
+  score = 0;
+  updateScoreValue();
+
+  startButton.textContent = 'Restart';
+
+  hideStartMessage();
+  updateStartButton();
+  setupInputOnce();
+}
+
+function hideStartMessage() {
+  const startMessage = document.querySelector('.message-start');
+
+  startMessage.classList.add('hidden');
+}
+
+function updateStartButton(gameOver = false) {
+  startButton.textContent = gameOver ? 'Start' : 'Restart';
+
+  if (!gameOver) {
+    startButton.classList.add('restart');
+  } else {
+    startButton.classList.remove('restart');
+  }
+
+  startButton.addEventListener('click', startGame);
+}
+
+function updateScoreValue(value = 0) {
+  score += value;
+  scoreElement.textContent = `${score}`;
+}
+
+function setupInputOnce() {
+  window.addEventListener('keydown', handleInput, { once: true });
+}
+
+function handleInput(e) {
+  switch (e.key) {
+    case 'ArrowUp':
+      e.preventDefault();
+
+      if (!canUp()) {
+        setupInputOnce();
+
+        return;
+      }
+      moveUp();
+      break;
+    case 'ArrowDown':
```

Comment:
Move all cases to the object.

Example of usage:
`case Directions.ArrowDown:`